File: gc-starter/crud-spring-boot-starter/src/main/java/com/gc/starter/crud/query/PageQueryParameter.java
package com.gc.starter.crud.query;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * 分页查询类
 * @author jackson
 * 2020/3/14 6:55 下午
 */
@Getter
@Setter
@ToString
@EqualsAndHashCode(callSuper = true)
public class PageQueryParameter<K, V> extends SortQueryParameter<K, V> {
    private static final long serialVersionUID = -2738378597906345854L;

    private Integer limit;

    private Integer offset = 0;

    private Integer page;

    private String keyword;
}

File: smart-starter/crud-spring-boot-starter/src/main/kotlin/com/smart/starter/crud/query/PageQueryParameter.kt
package com.smart.starter.crud.query


/**
 * 分页查询参数
 * @author ming
 * 2019/9/13 下午6:00
 */
open class PageQueryParameter : SortQueryParameter() {

    var limit: Int? = null

    var offset: Int = 0

    // 关键字查询
    var keyword: String? = null

    override fun toString(): String {
        return "${super.toString()}, limit=$limit, offset=$offset)"
    }

}